snapcraft/internal/parts.py

```diff
@@ -270,7 +270,8 @@ def load_plugin(self, part_name, plugin_name, part_properties):
             plugin_name=plugin_name,
             part_properties=part_properties,
             project_options=self._project_options,
-            part_schema=self._validator.part_schema)
+            part_schema=self._validator.part_schema,
+            definitions_schema=self._validator.definitions_schema)
```

Comment:
let's use yaml, I don't see us changing this any time soon. If you want to do the work, yes please.